engine/src/test/java/io/zeebe/engine/processing/bpmn/multiinstance/MultiInstanceCallActivityTest.java

```diff
@@ -185,6 +195,68 @@ public void shouldCancelChildInstancesOnTermination() {
             BpmnElementType.PROCESS);
   }
 
+  @Test
+  public void shouldCollectOutputFromChildInstance() {
+    // given
+    final BpmnModelInstance parentWorkflow =
+        parentWorkflowWithCallActivity(
+            callActivity ->
+                callActivity
+                    .zeebeOutputExpression("x", "result")
+                    .multiInstance(
+                        b ->
+                            b.zeebeInputCollectionExpression(INPUT_COLLECTION_VARIABLE)
+                                .zeebeOutputElementExpression("result")
+                                .zeebeOutputCollection("results")));
+
+    ENGINE.deployment().withXmlResource("wf-parent.bpmn", parentWorkflow).deploy();
+
+    final long workflowInstanceKey =
+        ENGINE
+            .workflowInstance()
+            .ofBpmnProcessId(PROCESS_ID_PARENT)
+            .withVariable(INPUT_COLLECTION_VARIABLE, INPUT_COLLECTION)
+            .create();
+
+    // when
+    awaitJobsCreated(INPUT_COLLECTION.size());
+
+    final var jobCounter = new AtomicInteger();
+
+    ENGINE
+        .jobs()
+        .withType(jobType)
+        .activate()
+        .getValue()
+        .getJobKeys()
+        .forEach(
+            jobKey ->
+                ENGINE
+                    .job()
+                    .withKey(jobKey)
+                    .withVariable("x", jobCounter.incrementAndGet())
+                    .complete());
+
+    RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_COMPLETED)
+        .withWorkflowInstanceKey(workflowInstanceKey)
+        .withElementType(BpmnElementType.MULTI_INSTANCE_BODY)
+        .await();
+
+    // then
+    final var expectedOutputCollection =
+        JsonUtil.toJson(
+            IntStream.rangeClosed(1, jobCounter.get()).boxed().collect(Collectors.toList()));
```

Comment:
It might be better to use `INPUT_COLLECTION.size()` instead of `jobCounter.get()`, to make the expected part less 'dynamic'.

Reply:
Ok :+1: